OpenRA.Mods.Cnc/Traits/Chronoshiftable.cs

```diff
@@ -59,14 +59,18 @@ public Chronoshiftable(ActorInitializer init, ChronoshiftableInfo info)
 			: base(info)
 		{
 			self = init.Self;
-			ReturnTicks = init.GetValue<ChronoshiftReturnInit, int>(info, 0);
-			duration = init.GetValue<ChronoshiftDurationInit, int>(info, 0);
-			Origin = init.GetValue<ChronoshiftOriginInit, CPos>(info, CPos.Zero);
-
-			// Defer to the end of tick as the lazy value may reference an actor that hasn't been created yet
-			var chronosphereInit = init.GetOrDefault<ChronoshiftChronosphereInit>(info);
-			if (chronosphereInit != null)
-				init.World.AddFrameEndTask(w => chronosphere = chronosphereInit.Value(init.World).Value);
+
+			var returnInit = init.GetOrDefault<ChronoshiftReturnInit>(info);
+			if (returnInit != null)
+			{
+				ReturnTicks = returnInit.Ticks;
+				duration = returnInit.Duration;
+				Origin = returnInit.Origin;
+
+				// Defer to the end of tick as the lazy value may reference an actor that hasn't been created yet
+				if (returnInit.Chronosphere != null)
+					init.World.AddFrameEndTask(w => chronosphere = returnInit.Chronosphere.Actor(init.World).Value);
```

Comment:
Something is still not working correctly here, I tried the following in allies 01 (just put it at the end of WorldLoaded):
```
	local cron = Actor.Create("PDOX", true, { Owner = player, Location = BaseCameraPoint.Location })
	Trigger.AfterDelay(2, function() Actor.Create("ca", true, { Owner = player, Location = Map.NamedActor("CruiserPoint2").Location, ChronoshiftReturn = { Origin = Map.NamedActor("CruiserPoint4").Location, Ticks = 100, Duration = 50, Chronosphere = cron }}) end)
```
and that crashes with
```
Exception of type `System.Collections.Generic.KeyNotFoundException`: Der angegebene Schlüssel war nicht im Wörterbuch angegeben.
   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at OpenRA.Mods.Common.ActorInitActorReference.InnerValue(World world) in OpenRA\OpenRA.Mods.Common\ActorInitializer.cs:Line 114.
   at OpenRA.Mods.Common.ActorInitActorReference.<>c__DisplayClass5_0.<Actor>b__0() in OpenRA\OpenRA.Mods.Common\ActorInitializer.cs:Line 124.
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at OpenRA.Mods.Cnc.Traits.Chronoshiftable.<>c__DisplayClass7_1.<.ctor>b__0(World w) in OpenRA\OpenRA.Mods.Cnc\Traits\Chronoshiftable.cs:Line 72.
   at OpenRA.World.Tick() in OpenRA\OpenRA.Game\World.cs:Line 467.
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in OpenRA\OpenRA.Game\Game.cs:Line 602.
   at OpenRA.Game.LogicTick() in OpenRA\OpenRA.Game\Game.cs:Line 629.
   at OpenRA.Game.Loop() in OpenRA\OpenRA.Game\Game.cs:Line 793.
   at OpenRA.Game.Run() in OpenRA\OpenRA.Game\Game.cs:Line 834.
   at OpenRA.Game.InitializeAndRun(String[] args) in OpenRA\OpenRA.Game\Game.cs:Line 262.
   at OpenRA.Program.Main(String[] args) in OpenRA\OpenRA.Game\Support\Program.cs:Line 37.
```
as apparently the actor reference init is looking for a map placed actor (why?) instead of the actor I gave it.
When I don't give a `Chronosphere` argument, it doesn't crash when creating the init, but instead when the timer runs out and it tries to teleport back using a `null` chronosphere.

Reply:
My latest rewrite may have regressed the lua handling - will look into it.